### PR TITLE
#8210: support PixieBrix API in starter mod

### LINKED ISSUE #8210: Can't use mods using `@pixiebrix/api` as starter mods
## Describe the bug

- Can't use mods using `@pixiebrix/api` as starter mods because that integration configuration is not being returned as "built-in"
- The error then prevents all starter mods from being activated for the user
- Prevents us from marking the writing assist and snippet manager mods
- Related code: https://github.com/pixiebrix/pixiebrix-extension/blob/c295c6c99b4851c9a52807ac5199b36789d62824/src/background/starterMods.ts#L149-L149

## To Reproduce

1. Mark a mod using the `@pixiebrix/api` as a starter mod
2. Add a brick that makes a PixieBrix API call. To do so, you have to add a brick that accepts the `@pixiebrix/api` integration type. Our HTTP Request brick doesn't include the PixieBrix API as an option. The database bricks for example, accept a  `@pixiebrix/api`  configuration
3. Visit the welcome page

## Actual behavior

No starter mods activate

## Expected behavior

All the starter mods activate

## Screenshots/Loom

If applicable, add screenshots/looms to help explain your problem.

## Desktop (please complete the following information):

- The bug is also on main
- Browser: Edge
- Extension Version: 1.8.12-mv3 

## Workaround

- Set the integration as optional (via the workshop) so it passes the starter mod integration configuration check

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #8210 

## Checklist

- [x] Add jest or playwright tests and/or storybook stories
- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: src/background/starterMods.ts [hunk 1]
```diff
@@ -150,6 +151,13 @@ async function activateMods(modDefinitions: ModDefinition[]): Promise<boolean> {
 
   const builtInDependencies = unconfiguredIntegrationDependencies.map(
     (unconfiguredDependency) => {
+      if (unconfiguredDependency.integrationId === PIXIEBRIX_INTEGRATION_ID) {
+        return {
+          ...unconfiguredDependency,
+          configId: null,
```
Comment (BLoe) [inline src/background/starterMods.ts:157]:
```suggestion
          configId: PIXIEBRIX_INTEGRATION_CONFIG_ID,
```

File: src/background/starterMods.ts [hunk 2]
```diff
@@ -46,6 +46,7 @@ import { selectModComponentsForMod } from "@/store/extensionsSelectors";
 import { type SidebarState } from "@/types/sidebarTypes";
 import { getEventKeyForPanel } from "@/store/sidebar/eventKeyUtils";
 import { type UUID } from "@/types/stringTypes";
+import { PIXIEBRIX_INTEGRATION_ID } from "@/integrations/constants";
```
Comment (BLoe) [inline src/background/starterMods.ts:49]:
```suggestion
import { PIXIEBRIX_INTEGRATION_ID, PIXIEBRIX_INTEGRATION_CONFIG_ID } from "@/integrations/constants";
```

File: src/background/starterMods.ts [hunk 3]
```diff
@@ -150,6 +151,13 @@ async function activateMods(modDefinitions: ModDefinition[]): Promise<boolean> {
 
   const builtInDependencies = unconfiguredIntegrationDependencies.map(
     (unconfiguredDependency) => {
+      if (unconfiguredDependency.integrationId === PIXIEBRIX_INTEGRATION_ID) {
+        return {
```
Comment (BLoe) [inline src/background/starterMods.ts:155]: You could also just return `pixiebrixIntegrationDependencyFactory()` here 🤷‍♂️

Comment (twschiller) [reply src/background/starterMods.ts:159]: I added a comment, I think it's easier to just only replace the config property

**Review by BLoe — APPROVED** [review]

**Pull request merged**